backend/src/modules/idim-webservice/idim-webservice.service.ts

```diff
@@ -0,0 +1,131 @@
+                },
+                HttpStatus.BAD_REQUEST
+            );
+        }
+
+        // add autorization header for making the soap api call
+        const auth =
+            'Basic ' +
+            Buffer.from(
+                this.idimWebServiceUsername + ':' + this.idimWebServicePassword
+            ).toString('base64');
+
+        try {
+            const client = await soap.createClientAsync(
+                this.idimWebServiceUrl,
+                {
+                    wsdl_headers: { Authorization: auth },
+                }
+            );
+            client.addHttpHeader('Authorization', auth);
+
+            // set xml schema parameters for the request
+            const requestData = {
+                internalAccountSearchRequest: {
+                    onlineServiceId: this.idimWebServiceID,
+                    // who is sending the request
+                    requesterAccountTypeCode,
+                    requesterUserId,
+                    // some config parameter
+                    pagination: { pageSizeMaximum: '20', pageIndex: '1' },
+                    sort: { direction: 'Ascending', onProperty: 'UserId' },
+                    // who we search for, exact match userID
+                    accountMatch: {
+                        userId: {
+                            value: userId,
+                            matchPropertyUsing: 'Exact',
+                        },
+                    },
+                },
+            };
+
+            return new Promise((resolve, reject) => {
+                client.BCeIDService.BCeIDServiceSoap.searchInternalAccount(
+                    requestData,
+                    function (error, foundUser) {
+                        if (error) {
+                            reject(
+                                new HttpException(
+                                    {
+                                        error:
+                                            'IDIM web service call error: ' +
+                                            error,
+                                    },
+                                    HttpStatus.INTERNAL_SERVER_ERROR
+                                )
+                            );
+
+                            return null;
+                        }
+
+                        if (
+                            foundUser.searchInternalAccountResult.code ==
+                            'Failed'
+                        ) {
+                            reject(
+                                new HttpException(
+                                    {
+                                        status: HttpStatus.BAD_REQUEST,
+                                        ...foundUser.searchInternalAccountResult,
+                                    },
+                                    HttpStatus.BAD_REQUEST
+                                )
+                            );
+
+                            return null;
```

Comment:
Not sure if I am right, but same comment as above for returning null.

Reply:
changed to return reject